diff --git a/lmcache/v1/storage_backend/nixl_storage_backend.py b/lmcache/v1/storage_backend/nixl_storage_backend.py
index c4d1230..99ddff4 100644
--- a/lmcache/v1/storage_backend/nixl_storage_backend.py
+++ b/lmcache/v1/storage_backend/nixl_storage_backend.py
@@ -99,7 +99,7 @@ class NixlStorageConfig:
             else:
                 return False
         else:
-            if backend in ("GDS", "GDS_MT", "OBJ"):
+            if backend in ("GDS", "GDS_MT", "OBJ", "AIS", "AIS_MT"):
                 return device == "cpu" or device == "cuda"
             elif backend in ("POSIX", "HF3FS", "AZURE_BLOB"):
                 return device == "cpu"
@@ -219,7 +219,13 @@ class NixlDescPool(ABC):
 
 
 class NixlFilePool(NixlDescPool):
-    def __init__(self, size: int, path: str, use_direct_io: bool):
+    def __init__(
+        self,
+        size: int,
+        path: str,
+        use_direct_io: bool,
+        slot_size: int = 0,
+    ):
         super().__init__(size)
         self.fds: List[int] = []
 
@@ -238,6 +244,8 @@ class NixlFilePool(NixlDescPool):
             filename = f"obj_{i}_{uuid.uuid4().hex[0:4]}.bin"
             tmp_path = os.path.join(path, filename)
             fd = os.open(tmp_path, flags)
+            if slot_size > 0:
+                os.ftruncate(fd, slot_size)
             self.fds.append(fd)
 
     def close(self):
@@ -784,6 +792,7 @@ class NixlStaticStorageBackend(NixlStorageBackend):
             nixl_config.pool_size,
             nixl_config.path,
             nixl_config.use_direct_io,
+            slot_size=self.memory_allocator.align_bytes,
         )
         assert self.pool is not None
 
@@ -798,9 +807,15 @@ class NixlStaticStorageBackend(NixlStorageBackend):
         )
 
     @staticmethod
-    def createPool(backend: str, size: int, path: str, use_direct_io: bool):
-        if backend in ("GDS", "GDS_MT", "POSIX", "HF3FS"):
-            return NixlFilePool(size, path, use_direct_io)
+    def createPool(
+        backend: str,
+        size: int,
+        path: str,
+        use_direct_io: bool,
+        slot_size: int = 0,
+    ):
+        if backend in ("GDS", "GDS_MT", "POSIX", "HF3FS", "AIS", "AIS_MT"):
+            return NixlFilePool(size, path, use_direct_io, slot_size)
         elif backend in ("OBJ", "AZURE_BLOB"):
             return NixlObjectPool(size)
         else:
