diff --git a/lmcache/v1/storage_backend/local_cpu_backend.py b/lmcache/v1/storage_backend/local_cpu_backend.py
index c072465..a1a5c79 100644
--- a/lmcache/v1/storage_backend/local_cpu_backend.py
+++ b/lmcache/v1/storage_backend/local_cpu_backend.py
@@ -99,7 +99,10 @@ class LocalCPUBackend(AllocatorBackendInterface):
                 lmcache_worker=lmcache_worker,
             )
         else:
-            logger.warning("Controller message sender is not initialized")
+            if config.enable_controller:
+                logger.warning("Controller message sender is not initialized")
+            else:
+                logger.debug("Controller message sender is not initialized")
 
         self._setup_metrics()
 
