diff --git a/lmcache/v1/storage_backend/gds_backend.py b/lmcache/v1/storage_backend/gds_backend.py
index 83c31e7..c14eec5 100644
--- a/lmcache/v1/storage_backend/gds_backend.py
+++ b/lmcache/v1/storage_backend/gds_backend.py
@@ -1185,7 +1185,7 @@ class GdsBackend(AllocatorBackendInterface):
         Allocate a memory object of shape and dtype
         """
         if eviction:
-            logger.warning("GDS Backend does not support eviction")
+            logger.debug("GDS Backend does not support eviction")
 
         logger.debug("Allocating memory with busy loop: %s", busy_loop)
 
@@ -1235,7 +1235,7 @@ class GdsBackend(AllocatorBackendInterface):
         Batched allocate `batch_size` memory objects of shape and dtype
         """
         if eviction:
-            logger.warning("GDS Backend does not support eviction")
+            logger.debug("GDS Backend does not support eviction")
 
         logger.debug(
             "Batched allocating memory in GDS backend with busy loop: %s",
